cpp/src/gandiva/precompiled/time.cc

```diff
@@ -706,10 +706,14 @@ const char* castVARCHAR_timestamp_int64(int64 context, timestamp in, int64 lengt
   char char_buffer[char_buffer_length];
 
   // yyyy-MM-dd hh:mm:ss.sss
-  snprintf(char_buffer, char_buffer_length,
-           "%04" PRId64 "-%02" PRId64 "-%02" PRId64 " %02" PRId64 ":%02" PRId64
-           ":%02" PRId64 ".%03" PRId64,
-           year, month, day, hour, minute, second, millis);
+  int res = snprintf(char_buffer, char_buffer_length,
+                     "%04" PRId64 "-%02" PRId64 "-%02" PRId64 " %02" PRId64 ":%02" PRId64
+                     ":%02" PRId64 ".%03" PRId64,
+                     year, month, day, hour, minute, second, millis);
+  if (res < 0) {
+    // prevent warning on release build with gcc. should be unreachable.
+    return "";
```

Comment:
I didn't put `gdv_fn_context_set_error_msg` call here because this clause won't be executed.

Reply:
Should probably use a `DCHECK` here rather than this.

Reply:
I first tried to use `DCHECK(false)` but it couldn't prevent the warning because `DCHECK` vanishes on release build.  So, I'll put `DCHECK(false)` before `return ""`.

Reply:
Hmm, `DCHECK` in release mode is:
```cpp
#define DCHECK(condition)                     \
  while (false) ARROW_IGNORE_EXPR(condition); \
  while (false) ::arrow::util::detail::NullLog()
```

Reply:
I guess `DCHECK` family used from Gandiva is the following part:

https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/logging.h#L21-L37